### PR TITLE
Show custom color when pie chart is empty.

### PR DESCRIPTION
**Pull request by sharmadhiraj**
**Show custom color when pie chart is empty.**

Pass color as parameter emptyColor (optional), then pie chart will be of that color when total is empty.

Related issue:
https://github.com/apgapg/pie_chart/issues/35

Screenshot_20200919-123638

### CODE REVIEW

File: lib/src/chart_painter.dart [hunk 1]
```diff
@@ -51,58 +54,59 @@ class PieChartPainter extends CustomPainter {
...
     _subTitles = titles;
-    _total = values.fold(0, (v1, v2) => v1 + v2);
   }
 
   @override
   void paint(Canvas canvas, Size size) {
     final side = size.width < size.height ? size.width : size.height;
-    _prevAngle = this.initialAngle*math.pi/180;
-    for (int i = 0; i < _subParts.length; i++) {
+    if (emptyColor != null && _total == 0) {
```
Comment (apgapg) [inline lib/src/chart_painter.dart:64]: Do not explicitly check emptyColor value. Just make emptyColor having a default Color in constructor

**Comment by apgapg** [comment]
Hi @sharmadhiraj  Thats a cool and much needed addition.
There can be some improvements that can make this better.

You must give a default value to the piechart constructor for **emptyColor** hence in this way one doesnt explicity need to provide color and hence no emptyColor null check in code.

Hope you would do the needful and push the new changes

File: lib/src/chart_painter.dart (hunk 1 above)
Comment (sharmadhiraj) [reply lib/src/chart_painter.dart]: Fixed.

**Review by apgapg — APPROVED** [review]
LGTM

**Pull request merged**